internal/metadataproviders/system/metadata.go

```diff
@@ -128,6 +130,22 @@ func (p systemMetadataProvider) reverseLookup(ipAddresses []string) (string, err
 	return "", fmt.Errorf("reverseLookup failed to convert IP addresses to name: %w", err)
 }
 
-func (p systemMetadataProvider) HostID() (string, error) {
-	return machineid.ID()
+func (p systemMetadataProvider) HostID(ctx context.Context) (string, error) {
```

Comment:
Is this method tested anywhere? i don't see any specific tests around these changes.

Reply:
Not sure how to test it properly as extracting `resource` to interface for me looks like unnecessary complication. I am open for suggestions